File: irt_gui_4/src/main/java/irt/gui/web/beans/flash/FlashCommand.java
package irt.gui.web.beans.flash;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;

@AllArgsConstructor(access = AccessLevel.PRIVATE)
public enum FlashCommand {
	EMPTY			(new byte[] { }),
	/** DEC:{127}; HEX:{0x7F} */
	CONNECT			(new byte[] { 0x7F }),
	/** DEC:{0, 255}; HEX: {0x00, 0xFF} */
	GET				(new byte[] { 0x00, (byte) 0xFF }),
	/** DEC: {2,253}; HEX: { 0x02, 0xFD} */
	GET_ID			(new byte[] { 0x02, (byte) 0xFD }),
	/** DEC: {17,238}; HEX: { 0x11, 0xEE} */
	READ_MEMORY		(new byte[] { 0x11, (byte) 0xEE }),
	/** DEC: {49,206}; HEX: { 0x31, 0xCE} */
	WRITE_MEMORY	(new byte[] { 0x31, (byte) 0xCE }),
	/** DEC: {67,188}; HEX: { 0x43, 0xBC} */
	ERASE			(new byte[] { 0x43, (byte) 0xBC }),
	/** DEC: {68,187}; { 0x44, 0xBB} */
	EXTENDED_ERASE	(new byte[] { 0x44, (byte) 0xBB });

	private byte[] command;

	public byte[] toBytes() {
		return command;
	}

}

File: irt_gui_4/src/main/resources/static/js/flash/flash-command.js
class FlasCommand{

	#name;
	#bytes;
	constructor(name, bytes){
		this.#name = name;
		this.#bytes = bytes;
	}

	get name(){
		return this.#name;
	}
	get command(){
		return this.#bytes;
	}
}

const commands = [
	new FlasCommand('EMPTY'			, []),
	new FlasCommand('CONNECT'		, [0x7F]),
	new FlasCommand('GET'			, [0x00, 0xFF]),	// 0x00 - command; 0xFF - checksum]
	new FlasCommand('GET_VERSION'	, [0x01, 0xFE]),
	new FlasCommand('GET_ID'		, [0x02, 0xFD]),
	new FlasCommand('READ_MEMORY'	, [0x11, 0xEE]),
	new FlasCommand('WRITE_MEMORY'	, [0x31, 0xCE]),
	new FlasCommand('ERASE'			, [0x43, 0xBC]),
	new FlasCommand('EXTENDED_ERASE', [0x44, 0xBB])
];

export default commands;
export function getCommand(name){
	return commands.find(fc=>fc.name===name).command;
}
